### DISCUSSION TITLE
Builtin Features (import and cimport update)

### DISCUSSION THREAD

**Discussion by AniketKumar2500** [CsqLang/Csq#260] [Ideas]
I hope we nearly made everything which can be used to code like classes, functions etc.
To improve it we should also have already built features which an user directly use without coding it on it's own just like Python do and just like Go, Rust and Julia do.

We shall also try to write some libraries for different kind of tasks like data science(We can also use C/C++ for it and embed it using cimport) if we are doing so there is a probability that we can find hidden bugs and fix them making Csq bug free.

**Comment by WithSJ** [comment]
good

**Comment by WithSJ** [comment]
I propose organizing the project with a "Include" folder. Within "Include," there are two subfolders: "cimport" for .cpp files generated by CsqCore and "import" for .csq files comprising modules created by the csq lang.

When utilizing "cimport" or "import" in a csq code file, the Csq compiler follows a specific hierarchy. It first checks the current folder for the module. If the module is not found there, the compiler proceeds to check the "cimport" or "import" folder, respectively. If the module is still not located, the compiler returns an error.
